src/systems/filecoin_vm/runtime/impl/runtime.go

```diff
@@ -618,28 +630,20 @@ func (rt *VMContext) NewActorAddress() addr.Address {
 }
 
 func (rt *VMContext) IpldPut(x ipld.Object) ipld.CID {
-	var serializedSize int
-	IMPL_FINISH()
-	panic("") // compute serializedSize
-
-	rt._rtAllocGas(gascost.IpldPut(serializedSize))
-
-	IMPL_FINISH()
-	panic("") // write to IPLD store
+	IMPL_FINISH() // Serialization
+	serialized := []byte{}
+	cid := rt._store.Put(serialized)
+	rt._rtAllocGas(gascost.IpldPut(len(serialized)))
+	return cid
 }
 
-func (rt *VMContext) IpldGet(c ipld.CID) vmr.Runtime_IpldGet_FunRet {
-	IMPL_FINISH()
-	panic("") // get from IPLD store
-
-	var serializedSize int
-	IMPL_FINISH()
-	panic("") // retrieve serializedSize
-
-	rt._rtAllocGas(gascost.IpldGet(serializedSize))
-
-	IMPL_FINISH()
-	panic("") // return item
+func (rt *VMContext) IpldGet(c ipld.CID, o ipld.Object) bool {
+	serialized, ok := rt._store.Get(c)
+	if ok {
+		rt._rtAllocGas(gascost.IpldGet(len(serialized)))
```

Comment:
Where is `gascost` getting pulled in from? This is a random question but wouldn't you want to do the `rtAllocGas` in get even if the object isn't found? This is still a DoS vector...

Reply:
maybe there should be a gas cost for a failed lookup, yes. But it can't depend on the object's size.

Reply:
fair enough.